### ISSUE TITLE
AFNI build system

### ISSUE THREAD

**Issue by TheChymera** [afni/afni#25]
Hi, for a while I have been trying to package AFNI for Gentoo - see the old discussion on [afniHQ](https://github.com/afniHQ/AFNI/issues/3) and the long standing pull request of AFNI [on gentoo-science](https://github.com/gentoo-science/sci/pull/115).

A while ago, I heard setting up a build system (CMake?) is on the to-do list for the AFNI team. This would go a long way in helping create a sustainable package atoms on Gentoo (and many other distributions). Do you have any updates or a timeline on this?

**Comment by afni-rickr** [comment]
Yes, the Neurodebian folks are interested in getting this done, and they know how to do it (we do not).  We are planning to work on this together in the near future.

**Comment by nno** [comment]
I'll ping @yarikoptic and @hanke from the neurodebian project.

**Comment by yarikoptic** [comment]
yeap, @Alexqw might lead this effort unless someone else gets to it before... we are planing (no actions were done for a while due to lack of time/personnel to dedicate) to

1. update our neurodebian cmake setup so we build recent snapshot against our stripped/modularized (data lives outside etc) afni sources
2. update/extend cmake files to build stock afni sources, which would entail providing cmake'ification for 3rd party libraries embedded within afni source tree

we will update here whenever anything to update with ;)

**Comment by Alexqw** [comment]
The soonest I can start working on this will be mid-next-week. I'm juggling a few too many projects right now to give it the proper attention quite yet.

---Alex

**Comment by nno** [comment]
@TheChymera @yarikoptic, the AFNI developers (@afni-rickr @afni-rwcox) would like to permanently /delete/ the https://github.com/afnihq/afni (afniHQ) repository, because it has become quite redundant now that the https://github.com/afni/afni repository is used as the primary github location for AFNI. 

However you linked to the discussion thread at https://github.com/afnihq/afni (https://github.com/afniHQ/AFNI/issues/3). Is there any information there that you think should be preserved? If so, which parts, and do you think it would suffice to copy-paste the relevant parts in the current threat here? 

Would there be any other objections to deleting https://github.com/afnihq/afni (afniHQ)? Note that the user afniHQ will be preserved; only the repository https://github.com/afnihq/afni would be deleted.

**Comment by yarikoptic** [comment]
@nno ok to delete on my side -- here is a full copy of the discussion in pdf:  http://www.onerussian.com/tmp/PackagingAFNIforGentoo-3.pdf

**Comment by TheChymera** [comment]
@yarikoptic thank you. @nno I just linked as a reference, I thought it would be archived somehow rather than deleted. In any case the issue with the discussion is now resolved!

**Comment by nno** [comment]
@yarikoptic @TheChymera very good, thanks for responding quickly. It seems that in any case all the important information has been preserved (thanks @yarikoptic), and if the issue of the discussion has been resolved, it appears that afniHQ/afniHQ is now close to being deleted.

**Comment by afni-rickr** [comment]
It's dead, Jim.

**Comment by nno** [comment]
Excellent @afni-rickr, thanks for cleaning this up.

**Comment by nno** [comment]
Ping: @Alexqw @TheChymera as this may be relevant for you.

Thanks to recent efforts by @yarikoptic and @afni-rickr, AFNI now builds automatically after every push to master on travis: https://travis-ci.org/afni/afni

No cmake functionality (yet) as far as I know, but it's a first step towards publicly viewable, automated testing of the builds.

**Comment by yarikoptic** [comment]
@TheChymera so -- what have you end up with doing for building afni for gentoo?

**Comment by TheChymera** [comment]
@yarikoptic I ended up using FSL and nipy* only :-/ sad but true. I was hoping to get another two gentoo devs on board and finally fix this. But we haven't gotten around to taking a closer look at this. What about you?

**Comment by yarikoptic** [comment]
Never give up never give in... But sometimes get show and whiny :-) (eg https://github.com/afni/afni/issues/53)
